metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1548,10 +1550,13 @@ class MetalsLspService(
       params: SemanticTokensParams
   ): CompletableFuture[SemanticTokens] = {
     CancelTokens.future { token =>
-      compilers.semanticTokens(params, token).map { semanticTokens =>
-        if (semanticTokens.getData().isEmpty()) null
-        else semanticTokens
-      }
+      for {
+        _ <- userConfigPromise.future
```

Comment:
Potentially, we could have a similar issue with parent code lenses `superMethodLensesEnabled`, no? Though I think we might send refresh in that case :thinking: We might also send refresh for semantic tokens when they change.

Maybe we should wait for the build server promise, which should most likely be after requesting semantic tokens?

Reply:
Code lenses already wait for the `buildServerPromise` so that's fine.